### PR TITLE
Add icon without red dot for red theming colors

### LINKED ISSUE #104: automatic color adjustments should take place based on the selected theme color
As requested by @juliushaertl (https://github.com/nextcloud/server/issues/7441#issuecomment-350861648) I've created this issue.

## Expected behavior
Depending on the selected theme color the notification alert color, and maybe others, should automatically be adjusted to be still visible.

## Current behavior
No automatic color adjustments seems to take place at all.

### Steps to reproduce
1. Open the admin settings.
2. Change the theming color to `FF0000`.
   20171210_nc_theming_color

3. Check the notification symbol in the menu bar
20171210_nc_theming_alert_color

4. The notification alert is nearly invisible.

Better visibility could look like this:
20171210_nc_theming_alert_better_color

#### Server Configuration
OS: Linux 3.16.50
Web server: Apache2 2.4.29
Database: MariaDB 10.0.32
PHP version: 5.6.29
Nextcloud version: 12.0.4

#### Client Configuration
Browser: Mozilla Firefox 57.0.2
Operating system: Windows 7

### PR DESCRIPTION
**Pull request by juliusknorr**
This adds a fallback icon for primary colors that are red.

screenshot_2018-06-29 dateien - nextcloud

The fallback is not used on bright redish colors and on to dark colors where the red dot is clearly visible.

fixes #104 

@nextcloud/designers

### CODE REVIEW

**Comment by MariusBluem** [comment]
By testing this PR, I saw we are not showing the notification icon permanently ... is this by intention? I thought we wanted to change this behavior? @nickvergessen 

Ref.: https://github.com/nextcloud/notifications/pull/95 (PR), Issue: https://github.com/nextcloud/notifications/issues/2 🤔

**Comment by juliushaertl** [comment]
@MariusBluem Works fine for me here.

**Comment by nickvergessen** [comment]
Hmm, should we really pick white? Or use another color or black for the dot, so it's better spotable?

**Comment by MariusBluem** [comment]
We could simply take the opposite color:
b224934e-863f-4264-a13e-69aa21be1537

**Comment by juliushaertl** [comment]
The issue with a black dot is that it might not be nicely visible on the various brightness levels of red colors we can have there:

image image image

@MariusBluem Complementary colors also don't work nicely in terms of contrast:
image

The white one still seems to be the most obvious one to me, that is clearly visible though all red color values. In cases when the theming color is bright enough, we still use the regular black icon with the red dot, which also looks good then:

image

**Comment by nickvergessen** [comment]
Yes, it's "visible" but not "disturbing" enough:
Just look at your screenshot:
image

Do you immediately see that there is a notification?

**Comment by juliushaertl** [comment]
> Do you immediately see that there is a notification?

I'd say yes, since the icon also doesn't have any opacity as the other icons, but I welcome other ideas on how we can make it a more obvious.

**Comment by MariusBluem** [comment]
Or we make it flashing ...

**Comment by nickvergessen** [comment]
We have a little animation, but only for ~3secs. If you load the app in the background you miss it, anyway... lets continue with being productive

**Comment by jancborchardt** [comment]
I’d say it’s a good enhancement already.

In addition, in https://github.com/nextcloud/notifications/issues/104#issue-281366355, @j-ed did a mockup with a blue notification bubble color (we could use Nextcloud blue) which would be used in these cases. Maybe worth a try @juliushaertl?

But I’m also fine with it being just white there, as mentioned the opacity is also different.

**Comment by juliushaertl** [comment]
Here are some examples with the blue dot, but I also think that this doesn't work well in terms of contrast:

image image image

But I've no strong opinion on this, let's not invest to much time into an edgecase like this. I'd say we should merge the white one for now and we can still improve later if needed.

File: css/styles.scss [hunk 1]
```diff
@@ -19,6 +19,29 @@
...
+	}
+
+	@if (hue($color-primary) < 15deg or hue($color-primary) > 330deg)
+	and (saturation($color-primary) > 70%)
+	and (lightness($color-primary) > 25% and lightness($color-primary) < 80%)
+	and (luma($color-primary) > 0.1 and luma($color-primary) < 0.6) {
+		.notifications-button.hasNotifications {
+			background-repeat: no-repeat;
+			background-position: center center;
+			background-image: url('../img/notifications-new-fallback.svg?v=1');
```
Comment (MorrisJobke) [inline css/styles.scss]: This file does not exist 🙈

**Comment by MorrisJobke** [comment]
Beside that I'm fine with the white one 👍

File: css/styles.scss (hunk 1 above)
Comment (juliushaertl) [reply css/styles.scss]: Fixed.

**Comment by MorrisJobke** [comment]
I checked out this branch and changed the color to #FF0000 but it still shows the old avatar. Even after clearing the browser cache.

bildschirmfoto 2018-07-03 um 18 05 37

Even after clearing the SCSS caches via `occ maintenance:repair` it still shows the old one.

**Pull request closed without merging**